includes/settings/class-wcpdf-settings-debug.php

```diff
@@ -904,6 +906,233 @@ private function get_settings_sections(): array {
 			'numbers'  => __( 'Numbers', 'woocommerce-pdf-invoices-packing-slips' ),
 		) );
 	}
+	
+	/**
+	 * Fetch number table data
+	 *
+	 * @param string     $table_name
+	 * @param string     $orderby
+	 * @param string     $order
+	 * @param string|int $limit
+	 * @param integer    $chunk_size
+	 * @param integer    $offset
+	 * @param integer    $total_fetched
+	 * 
+	 * @return void
+	 */
+	public function fetch_number_table_data( string $table_name, string $orderby = 'id', string $order = 'desc', $limit = 'all', int $chunk_size = 100, int $offset = 0, int $total_fetched = 0 ): void {
+		global $wpdb;
+		
+		$input_data = array(
+			'table_name' => $table_name,
+			'orderby'    => $orderby,
+			'order'      => $order,
+			'limit'      => $limit,
+		);
+		
+		$data = $this->filter_fetch_request_data( $input_data );
+		
+		if ( empty( $data['table_name'] ) ) {
+			return;
+		}
+		
+		$total_limit   = 'all' !== $data['limit'] ? absint( $data['limit'] ) : 0;
+		$total_fetched = absint( $total_fetched ?? 0 );
+		$chunk_size    = 0 !== $total_limit && $total_limit < $chunk_size ? min( $total_limit - $offset, $chunk_size ) : absint( $chunk_size ?? 100 );
```

Comment:
Considering the `$total_limit` is `110`, and the default `$chunk_size` is `100`, for the first call to this method, it'll get 100 rows, and for the second call to this method, it'll get 100 rows instead of just 10, since `$total_limit < $chunk_size`.